### PR TITLE
feat: file/uuid path which returns file with original filename

### PR DESCRIPTION
**Pull request by flatypus**
Exposes a route at .../file/{uuid} that returns the file with the original filename. Includes a link field in the file details modal.

https://github.com/chibisafe/chibisafe/assets/68029599/f64cde80-f29c-46ec-aaeb-77a5ac1dd16a

### CODE REVIEW

File: packages/backend/src/main.ts [hunk 1]
```diff
@@ -252,6 +253,8 @@ const start = async () => {
 		await server.register(fstatic, {
 			root: fileURLToPath(new URL('../../../uploads', import.meta.url))
 		});
+
+		server.get('/file/:uuid', getOriginalFile);
```
Comment (flatypus) [inline packages/backend/src/main.ts:257]: I wasn't sure what the case should be if it's not serving /uploads, so I put the getOriginalFile line here...

File: packages/backend/src/utils/File.ts [hunk 2]
```diff
@@ -251,12 +252,15 @@ export const constructFilePublicLink = ({
...
 	req: FastifyRequest;
+	uuid?: string;
 }) => {
 	const host = SETTINGS.serveUploadsFrom ? SETTINGS.serveUploadsFrom : getHost(req);
 	const data = {
 		url: isS3
 			? `${SETTINGS.S3PublicUrl || SETTINGS.S3Endpoint}${quarantine ? '/quarantine' : ''}/${fileName}`
 			: `${host}${quarantine ? '/quarantine' : ''}${isWatched ? '/live' : ''}/${fileName}`,
+		// uuid url that accounts for s3
+		uuidUrl: uuid ? `${isS3 ? SETTINGS.S3PublicUrl || SETTINGS.S3Endpoint : host}/file/${uuid}` : '',
```
Comment (flatypus) [inline packages/backend/src/utils/File.ts:263]: also same here, what should be the case for isS3?

File: packages/backend/src/utils/GetOriginalFile.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { URL, fileURLToPath } from 'node:url';
```
Comment (flatypus) [inline packages/backend/src/utils/GetOriginalFile.ts:1]: I also wasn't sure what to call/where to put this route; putting it with the rest of the API routes doesn't feel right

**Comment by Pitu** [comment]
@flatypus thank you for the PR. After reviewing it and discussing with @iCrawl we decided to go with a simpler approach that ultimately achieves the same result in his PR https://github.com/chibisafe/chibisafe/pull/572

Sorry this didn't make it in 🙇‍♂️

**Pull request closed without merging**